config/decoder.go

```diff
@@ -231,6 +224,22 @@ func (d *decoder) mapping(childKey string, value reflect.Value, def string) erro
 	return nil
 }
 
+// Sets value to an interface type.
+func (d *decoder) iface(key string, value reflect.Value, def string) error {
+	v := d.getGlobalProvider().Get(key)
+	if !v.HasValue() || v.Value() == nil {
+		return nil
+	}
+
+	src := reflect.ValueOf(v.Value())
+	if src.Type().Implements(value.Type()) {
+		value.Set(src)
+		return nil
+	}
+
+	return fmt.Errorf("%v doesn't implement %v", src.Type(), value.Type())
```

Comment:
consider adding quotes around the %vs, either with %q or doing:

```go
return fmt.Errorf(`"%v" doesn't implement "%v"`, ...)
```

to make it clear which parts are constant and which parts are variant on the problematic type